Source/WebCore/css/CSSPrimitiveValue.h

```diff
@@ -177,7 +178,7 @@ class CSSPrimitiveValue final : public CSSValue {
 
     WEBCORE_EXPORT String stringValue() const;
 
-    const Color& color() const { ASSERT(primitiveUnitType() == CSSUnitType::CSS_RGBCOLOR); return *m_value.color; }
+    const StyleColor& color() const { ASSERT(primitiveUnitType() == CSSUnitType::CSS_RGBCOLOR); return *m_value.color; }
```

Comment:
StyleColor represents a post-style resolution (computed style) color in RenderStyle. It shouldn't be used in CSSValue (which is pre-style resolution). 'Currentcolor' on this level is represented by id value CSSValueCurrentcolor so a concrete color value works just fine at least for now. The mapping to StyleColor happens in Style::Builder and its helpers.

I don't think this patch needs to touch CSSPrimitiveValue at all. Making this change should make this patch way more tractable.